mat64/dense_test.go

```diff
@@ -1234,16 +1390,17 @@ func (s *S) TestOuter(c *check.C) {
 		want.Mul(xm, ym)
 
 		var m Dense
-		// Check with a new matrix
-		m.Outer(NewVector(len(test.x), test.x), NewVector(len(test.y), test.y))
-		c.Check(Equal(&m, want), check.Equals, true, check.Commentf("Test %v. Want %v, Got %v", i, want, m))
-		// Check with the same matrix
-		m.Outer(NewVector(len(test.x), test.x), NewVector(len(test.y), test.y))
-		c.Check(Equal(&m, want), check.Equals, true, check.Commentf("Test %v. Want %v, Got %v", i, want, m))
+		for j := 0; j < 2; j++ {
+			// Check with a new matrix - and then again.
+			m.Outer(NewVector(len(test.x), test.x), NewVector(len(test.y), test.y))
+			if !Equal(&m, want) {
+				t.Errorf("unexpected result for Outer test %d iteration %d: got: %+v want: %+v", i, j, m, want)
+			}
```

Comment:
That would be a logic change, and I didn't write the original test, so I don't want to do that. If you think there should have been a zero between the two calls (at L1240 in the original) then you should send a PR.